cmd/kaf/consume.go

```diff
@@ -370,21 +372,30 @@ func protoDecode(reg *proto.DescriptorRegistry, b []byte, _type string) ([]byte,
 		return b, nil
 	}
 
-	err := dynamicMessage.Unmarshal(b)
-	if err != nil {
-		// if there is an error, it might be a Confluent message, try parsing without headers
+	// user requested confluent aware decoding
+	if confluentHeader {
 		bTrimmed := discardConfluentHeader(b)
-		trimmedErr := dynamicMessage.Unmarshal(bTrimmed)
-		// if trimming doesn't help just return original error
-		if trimmedErr != nil {
+		err := dynamicMessage.Unmarshal(bTrimmed)
+		if err != nil {
 			return nil, err
 		}
+	} else {
+		err := dynamicMessage.Unmarshal(b)
+		if err != nil {
+			// if there is an error, it might be a Confluent message, try parsing without headers
```

Comment:
Rather than try to make an assumption here, should the error just be returned with an additional message like "this may be a confluent message, retry with `--confluent-header`"?

Reply:
yeah, you're right, having it be more magical just means that users will be less prepared when the magic case fails